### PR TITLE
chore(examples): add nycflights13 dataset to examples

### LINKED ISSUE #8718: feat(examples): add NYC flights data to R registry
### Is your feature request related to a problem?

We are looking for an example dataset to demonstrate Ibis-ML functionality. @jcrist suggested that adding the `nycflights13` data to Ibis core would be useful for demonstrating a number of other functionalities, such as time series features.

### Describe the solution you'd like

`ibis.examples.nycflights13.fetch()` :airplane:

### What version of ibis are you running?

8.0.0

### What backend(s) are you using, if any?

Any

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gforsyth**
Resolves #8718

As an example:

``` python
[ins] In [1]: from ibis.interactive import *

[ins] In [2]: ibis.examples.nycflights13_airports.fetch()
Out[2]: 
┏━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━┓
┃ faa    ┃ name                       ┃ lat       ┃ lon         ┃ alt   ┃ tz    ┃ … ┃
┡━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━┩
│ string │ string                     │ float64   │ float64     │ int64 │ int64 │ … │
├────────┼────────────────────────────┼───────────┼─────────────┼───────┼───────┼───┤
│ 04G    │ Lansdowne Airport          │ 41.130472 │  -80.619583 │  1044 │    -5 │ … │
│ 06A    │ Moton Field Municipal Air… │ 32.460572 │  -85.680028 │   264 │    -6 │ … │
│ 06C    │ Schaumburg Regional        │ 41.989341 │  -88.101243 │   801 │    -6 │ … │
│ 06N    │ Randall Airport            │ 41.431912 │  -74.391561 │   523 │    -5 │ … │
│ 09J    │ Jekyll Island Airport      │ 31.074472 │  -81.427778 │    11 │    -5 │ … │
│ 0A9    │ Elizabethton Municipal Ai… │ 36.371222 │  -82.173417 │  1593 │    -5 │ … │
│ 0G6    │ Williams County Airport    │ 41.467306 │  -84.506778 │   730 │    -5 │ … │
│ 0G7    │ Finger Lakes Regional Air… │ 42.883565 │  -76.781232 │   492 │    -5 │ … │
│ 0P2    │ Shoestring Aviation Airfi… │ 39.794824 │  -76.647191 │  1000 │    -5 │ … │
│ 0S9    │ Jefferson County Intl      │ 48.053809 │ -122.810644 │   108 │    -8 │ … │
│ …      │ …                          │         … │           … │     … │     … │ … │
└────────┴────────────────────────────┴───────────┴─────────────┴───────┴───────┴───┘
```

To do this I've added an entry to `gen_registry.py` that pulls the flight data and converts it to zstd-compressed parquet.

I've also added a `pixi.toml` and `pixi.lock` and a short README for the examples folder so other maintainers can start with a little more context.

I should also note that this is very far from perfect -- currently, when running `gen_registry.py`, the nycflights data DOES get added, but it also removes several entries from the `metadata.json` file.  I've manually removed all of those changes.

The short version, though, is that what `gen_registry` currently does only provides a subset of our current examples offering.

### CODE REVIEW

**Comment by gforsyth** [comment]
Also, the lock file is enormous and if we'd rather it didn't get checked in, I'm happy to remove it and to pin down the dependencies in the `pixi.toml` file a little more closely.

**Review by cpcloud — APPROVED** [review]
